packages/security_center/lib/routes.dart

```diff
@@ -110,11 +112,30 @@ enum Routes {
     return Routes.fromRoute(route).title(l10n, queryParameters);
   }
 
-  static YaruMasterTileBuilder get tileBuilder =>
-      (context, index, selected, availableWidth) => YaruMasterTile(
-            leading: Icon(values[index].icon?.call(selected)),
-            title: Text(values[index].title(AppLocalizations.of(context))),
+  static YaruMasterTileBuilder get tileBuilder => (
+        context,
+        index,
+        selected,
+        availableWidth,
+      ) =>
+          YaruMasterTile(
+            leading: Icon(availableRoutes[index].icon?.call(selected)),
+            title: Text(
+              availableRoutes[index].title(AppLocalizations.of(context)),
+            ),
           );
   static IndexedWidgetBuilder get pageBuilder =>
-      (context, index) => values[index].builder(context);
+      (context, index) => availableRoutes[index].builder(context);
+
+  static List<Routes> get availableRoutes {
+    final routes = <Routes>[Routes.appPermissions];
+
+    // Only include disk encryption if the feature is available
+    final featureService = getService<FeatureService>();
+    if (featureService.isDiskEncryptionAvailable) {
+      routes.add(Routes.diskEncryption);
+    }
+
+    return routes;
+  }
```

Comment:
At this point we should probably create a dedicated `AvailableRoutes` class and instantiate it once with a fixed set of routes. As a static method this currently checks whether disk encryption is available every time there is a widget rebuild. Doing this only once when the app starts should be enough.

[Screencast From 2025-07-18 14-39-23.webm](https://github.com/user-attachments/assets/e07aaca1-6d75-409b-9fcb-d19d2ecea9a9)